### ISSUE TITLE
Install Miniforge in a docker container

### ISSUE THREAD

**Issue by nikfio** [conda-forge/miniforge#617]
### Comment:

Hello everyone,

for pipeline design purposes, I would like to build an image based on Linux and one based on Windows.
I always use MIniforge in my development computer and what I want is try to innstall it also in containers.

While on a Linux based image everything runs smoothly, I can't install MIniforge on a Windows based container.

Here are the steps I went through:

1.  download a windows image with powershell

`docker pull m365pnp/powershell:1.10.0-nanoserver-1809`

3.  run the container and spawn a powershell

`docker run -it m365pnp/powershell:1.10.0-nanoserver-1809 pwsh`

5. download miniforge installation esecutable inside the container

`curl -L -O -k "https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Windows-x86_64.exe"`

7. Run installation

`Start-Process "C:\Miniforge3-Windows-x86_64.exe" -ArgumentList "/S /D=C:\Miniforge3" -NoNewWindow -Wait`

By launching the last step command It actually does nothing, no output and no installation performed.

Do you have any idea on the matter? Did you follow other ways to build a pipeline with windows based images?

Many thanks,
Nick

**Comment by jaimergp** [comment]
Maybe you are luckier with `cmd`? Check this [function in `constructor`](https://github.com/conda/constructor/blob/3bb0fd4434efa7770554e0e559efc09692dd0f62/tests/test_examples.py#L106-L130) or [this CI line in `napari`](https://github.com/napari/packaging/blob/aefac26631585ec84afb11477acf50adcf06ba6f/.github/workflows/make_bundle_conda.yml#L577).

**Comment by jakirkham** [comment]
Also would checkout [the miniforge-images repo]( https://github.com/conda-forge/miniforge-images )

**Comment by nikfio** [comment]
Hi guys,

Thank you for the suggestions.
I tried @jaimergp cmd option but it was not successful.

I went through with a not so good workaround:

1. downloaded Miniforge exe installer on my development PC
2. run miniforge command line installation with `"/S /D=C:\Miniforge3"` arguments
3. Copied the C:\Miniforge3 folder content inside the windows container
4. saved container as a new image
5. run the new image with interactve powershell, navigate to miniforge folder and then Scripts folder, run `conda.exe` init 
6. resaved to a new image with docker commit
7. Restarted the image and the command `mamba --version` gave output ok
8. Tested by running python and some mamba update commands


Hope it would help anyone in a similar situation.


Thanks,
Nicola
